include/net/nrf_cloud_coap.h

```diff
@@ -274,7 +276,19 @@ int nrf_cloud_coap_shadow_get(char *buf, size_t buf_len, bool delta);
 int nrf_cloud_coap_shadow_state_update(const char * const shadow_json);
 
 /**
- * @brief Update the device status in the shadow.
+ * @brief Update the device's "desired state" in the shadow via the state/desired CoAP resource.
+ * This is normally only used to silence a shadow delta that is incompatible with the device,
+ * by overwriting the invalid desired values with the reported values.
+ *
+ * @param[in]     shadow_json Null-terminated JSON string to be written to the device's shadow.
+ *
+ * @return 0 if the request succeeded, a positive value indicating a CoAP result code,
```

Comment:
is `a positive value indicating a CoAP result code` always an error?  if so can you make that clear, here and in the other spots?